### ISSUE TITLE
Is it possible for pyright to report "unhashable type error"

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4145]
### Discussed in https://github.com/microsoft/pyright/discussions/4143

<div type='discussions-op-text'>

<sup>Originally posted by **daya0576** November  5, 2022</sup>
```python
from dataclasses import dataclass


@dataclass(frozen=False)
class CustomerModel:
    id: int
    name: str


d1 = {{}: "foo"}
    # TypeError: unhashable type: 'dict'
d2 = {CustomerModel(0, name="bar"): 100}  #
    # TypeError: unhashable type: 'CustomerModel'
```

Is it possible for pyright to report the unhashable type error in static checking, as the `__hash__` method returns `None`.</div>

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.279, which I just published. It will also be included in a future version of pylance.
